flow/include/flow/ThreadHelper.actor.h

```diff
@@ -743,7 +743,101 @@ Future<T> safeThreadFutureToFutureImpl(ThreadFuture<T> threadFuture) {
+	Promise<Void> ready;
+	Future<Void> onReady = ready.getFuture();
+	UtilCallback<Standalone<T>>* callback = new UtilCallback<Standalone<T>>(threadFuture, ready.extractRawPointer());
+	int unused = 0;
+	threadFuture.callOrSetAsCallback(callback, unused, 0);
+	try {
+		wait(onReady);
+	} catch (Error& e) {
+		// broken_promise can be thrown if the network is already shut down
+		ASSERT(e.code() == error_code_operation_cancelled || e.code() == error_code_broken_promise);
+		// prerequisite: we have exclusive ownership of the threadFuture
+		if (e.code() == error_code_operation_cancelled) {
+			threadFuture.cancel();
+		}
+		throw e;
+	}
+	// threadFuture should be ready
+	ASSERT(threadFuture.isReady());
+	if (threadFuture.isError())
+		throw threadFuture.getError();
+	if (BUGGIFY) {
+		return Standalone<T>(threadFuture.get(), Arena());
+	}
+	return threadFuture.get();
+}
+
+ACTOR template <class T>
+Future<Optional<Standalone<T>>> safeThreadFutureToFutureImpl(ThreadFuture<Optional<Standalone<T>>> threadFuture) {
+	Promise<Void> ready;
+	Future<Void> onReady = ready.getFuture();
+	UtilCallback<Optional<Standalone<T>>>* callback =
+	    new UtilCallback<Optional<Standalone<T>>>(threadFuture, ready.extractRawPointer());
+	int unused = 0;
+	threadFuture.callOrSetAsCallback(callback, unused, 0);
+	try {
+		wait(onReady);
+	} catch (Error& e) {
+		// broken_promise can be thrown if the network is already shut down
+		ASSERT(e.code() == error_code_operation_cancelled || e.code() == error_code_broken_promise);
+		// prerequisite: we have exclusive ownership of the threadFuture
+		if (e.code() == error_code_operation_cancelled) {
+			threadFuture.cancel();
+		}
+		throw e;
+	}
+	// threadFuture should be ready
+	ASSERT(threadFuture.isReady());
+	if (threadFuture.isError())
+		throw threadFuture.getError();
+	if (BUGGIFY) {
+		if (threadFuture.get().present()) {
+			return Standalone<T>(threadFuture.get().get(), Arena());
+		} else {
+			return Optional<Standalone<T>>();
+		}
+	}
+	return threadFuture.get();
+}
+
+// This safeThreadFutureToFutureImpl(Future) actor should be used only with BUGGIFY.
+// It simulates the case where the returned Standalone points to memory owned by `future`.
+// When `future` goes out of scope, the memory pointed to by the Standalone will be invalid.
+ACTOR
+template <typename T>
+Future<Standalone<T>> faultySafeThreadFutureToFutureImpl(Future<Standalone<T>> future) {
+	try {
+		Standalone<T> _ = wait(future);
+		return Standalone<T>(future.get(), Arena());
+	} catch (Error& e) {
+		throw e;
+	}
+}
+
+// This safeThreadFutureToFutureImpl(Future) actor should be used only with BUGGIFY.
+// It simulates the case where the returned Optional<Standalone> points to memory owned by `future`.
+// When `future` goes out of scope, the memory pointed to by the Optional<Standalone> will be invalid.
+ACTOR
+template <typename T>
+Future<Optional<Standalone<T>>> faultySafeThreadFutureToFutureImpl(Future<Optional<Standalone<T>>> future) {
```

Comment:
This version isn't faulty, exactly, but it's rather simulating the behavior of DLTransaction. Maybe we could call this something like `removeArenaFromStandalone(Future<Standalone<T>>())` and then comment that we are using it to simulate the behavior of the `DLApi`.

I also think we could likely use these implementations for the thread safe versions, so the thread safe ones would return something like:

```
if (BUGGIFY) {
    return removeArenaFromStandalone(safeThreadFutureToFutureImpl(threadFuture));
}
```

and the future ones would return:

```
if (BUGGIFY) {
    return removeArenaFromStandalone(future);
}
```

That way we could avoid the mostly duplicate functions for `safeThreadFutureToFutureImpl`